File: sentimental-credit/credit.py
from cs50 import get_string

sum = 0
number = get_string("Number: ")
isValid = False
isFirst = True

if len(number) % 2 == 0:
    isFirst = True
else:
    isFirst = False

for i in range(0, len(number)):
    if isFirst:
        product = int(number[i]) * 2
        if product > 9:
            sum += int(str(product)[0])
            sum += int(str(product)[1])
        else:
            sum += product
        isFirst = False
    else:
        sum += int(number[i])
        isFirst = True

if sum % 10 == 0:
    isValid = True

if isValid:
    lastTwoD = number[0] + number[1]
    if (lastTwoD == "34" or lastTwoD == "37") and len(number) == 15:
        print("AMEX")
    elif (lastTwoD == "51" or lastTwoD == "52" or lastTwoD == "53" or lastTwoD == "54" or lastTwoD == "55") and len(number) == 16:
        print("MASTERCARD")
    elif lastTwoD[0] == "4" and (len(number) == 13 or len(number) == 16):
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number;
    long numberCopy;
    int counter = 0;
    int sum = 0;
    bool isValid = false;

    do
    {
        number = get_long("Number: ");
    } while(number <= 0);

    numberCopy = number;
    while(numberCopy != 0)
    {
        numberCopy /= 10;
        counter++;
    }


    for(int i = 0; i < counter; i++)
    {
        long tens = 1;
        for(int j = 0; j < i + 1; j++) {
            tens *= 10;
        }

        if(i % 2 != 0) {
            int product = number % tens / (tens / 10) * 2;
            if(product > 9) {
                sum += product % 10;
                sum += product / 10;
            } else {
                sum += product;
            }
        } else {
            sum += number % tens / (tens / 10);
        }
    }

    if(sum % 10 == 0) {
        isValid = true;
    }

    if(isValid) {
        long tens = 1;
        for(int i = 0; i < counter - 2; i++) {
            tens *= 10;
        }
        if((number / tens == 34 || number / tens == 37) && counter == 15) {
            printf("AMEX\n");
        } else if((number / tens == 51 || number / tens == 52 || number / tens == 53 || number / tens == 54 || number / tens == 55) && counter == 16) {
            printf("MASTERCARD\n");
        } else if(number / (tens * 10) == 4 && (counter == 13 || counter == 16)) {
            printf("VISA\n");
        } else {
            printf("INVALID\n");
        }
    } else {
        printf("INVALID\n");
    }
}
